spark/src/test/scala/org/apache/spark/sql/CometTPCDSQuerySuite.scala

```diff
@@ -158,6 +158,11 @@ class CometTPCDSQuerySuite
     conf.set(CometConf.COMET_EXEC_ALL_OPERATOR_ENABLED.key, "true")
     conf.set(CometConf.COMET_EXEC_SHUFFLE_ENABLED.key, "true")
     conf.set(CometConf.COMET_MEMORY_OVERHEAD.key, "20g")
+    conf.set(CometConf.COMET_SHUFFLE_ENFORCE_MODE_ENABLED.key, "true")
+    conf.set("spark.sql.adaptive.coalescePartitions.enabled", "true")
+    // Disable `CometTakeOrderedAndProjectExec` because it doesn't produce same output order
+    // as Spark.
+    conf.set("spark.comet.exec.takeOrderedAndProjectExec.disabled", "true")
```

Comment:
Is this a correctness issue in general or is the issue specific to this test because we have queries that are non-deterministic?

Reply:
I think these tests should be deterministic (that's why we can compare it with golden files). I'm not sure why `CometTakeOrderedAndProjectExec` returns out of order results.

The results are same, but the orders are different to Spark. I suspect that it is something related to sorting part in `CometTakeOrderedAndProjectExec`. As the sorting is delegated to DataFusion's sort/top k operators, I need to investigate particularly for the failed query (e.g., q6).

It is not related to the change here, though. So I will investigate it separately in follow PRs.

Reply:
Hmm, I found that the order of returned rows are not related to the sorting columns. For example, if you sort on [a, b], there are some rows with same [a, b] so the order of these rows are irrelevant to the sorting order, i.e., either

```
|a|b|c|
|0|1|1|
|0|1|2|
```

or 

```
|a|b|c|
|0|1|2|
|0|1|1|
```

are correct results for sorting on [a, b]. In these failed queries, Spark sort produces one and DataFusion sort produces another one. Because we compare the results as string, they are considered incorrect now.

Reply:
We maybe need to address this difference when comparing Spark and Comet query results.

Reply:
Currently I opened https://github.com/apache/datafusion-comet/pull/628 to enforce sorting for all join types.

Reply:
After #628, we don't need to disable `CometTakeOrderedAndProjectExec`.